### PR TITLE
Append Operator to LinearAdvectionLaxFriedrichs, Concatenation and ComponentProjection

### LINKED ISSUE #989: Inconsistencies in class naming
Here are examples of classes inheriting from `SomeInterface` and not called `...SomeInterface` (except `Operator` -> `Functional` and `VectorArray` -> `VectorArrayView`):

- `VectorSpace` (similar for `VectorArray`)
    - [`MPIVectorSpaceNoComm`](https://github.com/pymor/pymor/blob/c38840e05206990ba807887eeb10494f03b73453/src/pymor/vectorarrays/mpi.py#L252)
    - [`MPIVectorSpaceAutoComm`](https://github.com/pymor/pymor/blob/c38840e05206990ba807887eeb10494f03b73453/src/pymor/vectorarrays/mpi.py#L312)
- `Operator`
    - [`LinearAdvectionLaxFriedrichs`](https://github.com/pymor/pymor/blob/c38840e05206990ba807887eeb10494f03b73453/src/pymor/discretizers/builtin/fv.py#L448)
    - [`Concatenation`](https://github.com/pymor/pymor/blob/c38840e05206990ba807887eeb10494f03b73453/src/pymor/operators/constructions.py#L275)
    - [`ComponentProjection`](https://github.com/pymor/pymor/blob/c38840e05206990ba807887eeb10494f03b73453/src/pymor/operators/constructions.py#L629)

Should we rename some of them?

### PR DESCRIPTION
**Pull request by pmli**
Closes #989.

### CODE REVIEW

**Comment by pmli** [comment]
> Maybe it would be nice to have functions `Concatenation` etc. which are marked with `@deprecated` for the next release.

Agreed. Does it look ok?

**Comment by renefritze** [comment]
Should we maybe start to a running list of deprecations in the release notes? Would make writing them a bit simlper.

**Comment by pmli** [comment]
> Should we maybe start to a running list of deprecations in the release notes? Would make writing them a bit simlper.

Sounds good to me. I've added a remark to the release notes.

**Comment by sdrave** [comment]
> > Should we maybe start to a running list of deprecations in the release notes? Would make writing them a bit simlper.
> 
> Sounds good to me. I've added a remark to the release notes.

Can't we just grep for `@deprecated`?

File: src/pymor/basic.py [hunk 1]
```diff
@@ -44,11 +45,12 @@
...
-from pymor.discretizers.builtin.grids.boundaryinfos import EmptyBoundaryInfo, GenericBoundaryInfo, AllDirichletBoundaryInfo
+from pymor.discretizers.builtin.grids.boundaryinfos import (EmptyBoundaryInfo, GenericBoundaryInfo,
+                                                            AllDirichletBoundaryInfo)
 
-from pymor.operators.constructions import (LincombOperator, Concatenation, ComponentProjection, IdentityOperator,
-                                           ConstantOperator, ZeroOperator, VectorArrayOperator, VectorOperator,
-                                           VectorFunctional, FixedParameterOperator, AdjointOperator,
+from pymor.operators.constructions import (LincombOperator, ConcatenationOperator, ComponentProjectionOperator,
+                                           IdentityOperator, ConstantOperator, ZeroOperator, VectorArrayOperator,
+                                           VectorOperator, VectorFunctional, FixedParameterOperator, AdjointOperator,
```
Comment (sdrave) [inline src/pymor/basic.py:53]: I think, we should also add the deprecated versions for now, to ensure user code that imports from `basic` does not break.

**Review by sdrave — CHANGES_REQUESTED** [review]

**Review by sdrave — APPROVED** [review]

**Pull request merged**